### PR TITLE
[MM-40712] config: Bugfix on path resolution; fail if given config does not exist

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Now we fail if there is no file created for the `config.FileStore`.

#### Discussion
This implementation adds a new function argument as `createFileIfNotExists` into `config.NewStoreFromDSN` function which breaks the abstraction. We can only use this argument in `config.NewFileStore` only if we use this function in other layers.

In most cases we use `config.NewStoreFromDSN` hence I added the flag to `config.NewStoreFromDSN` function. I'm open to improve this further by using the new argument only in `config.NewFileStore`. But then we will need to decide whether we are going to create a store backend for file or db in the Api/App layers.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-40712

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: config/file.go [hunk 1]
```diff
@@ -66,7 +74,9 @@ func resolveConfigFilePath(path string) (string, error) {
 	// Otherwise, search for the config/ folder using the same heuristics as above, and build
 	// an absolute path anchored there and joining the given input path (or plain filename).
 	if configFolder, found := fileutils.FindDir("config"); found {
-		return filepath.Join(configFolder, path), nil
+		if configFile := fileutils.FindFile(filepath.Join(configFolder, path)); configFile != "" {
```
Comment (agnivade) [inline config/file.go:77]: Isn't this the same as L64? Wondering what's the difference here.

File: config/file.go [hunk 2]
```diff
@@ -30,7 +30,15 @@ type FileStore struct {
 }
 
 // NewFileStore creates a new instance of a config store backed by the given file path.
-func NewFileStore(path string) (fs *FileStore, err error) {
+func NewFileStore(path string, createFileIfNotExists bool) (fs *FileStore, err error) {
+	if createFileIfNotExists {
+		file, err2 := os.OpenFile(path, os.O_RDONLY|os.O_CREATE, 0644)
```
Comment (agnivade) [inline config/file.go:35]: Should we not do this in the `resolveConfigFilePath` function itself? The path passed here is not necessarily the final resolved path.

I am wondering if the check should be placed after the `resolvedPath` is computed. And then we check:

```go
  f, err := os.Open(resolvedPath)
  if err != nil && errors.Is(err, fs.ErrNotExist) {
    if createFileIfNotExists {
     // create file
    } else {
     return err
    }
   f.Close()
  }

```

And if we are doing this after `resolveConfigFilePath`, there might not be a need to modify it.

Comment (isacikgoz) [reply config/file.go:35]: Makes sense, will update accordingly.

File: config/file.go (hunk 1 above)
Comment (isacikgoz) [reply config/file.go:77]: Reverted it. It was checking if the parent config folder exits, rather than searching file itself.

**Review by agnivade — APPROVED** [review]
Nicely done :+1:

**Review by furqanmlk — APPROVED** [review]
image

**Pull request merged**